docs/store/api.md

```diff
@@ -20,6 +20,27 @@ import { reducers } from './reducers';
 export class AppModule {}
 ```
 
+### Initial State and Ahead of Time (AoT) Compilation
+
+Angular AoT requires all symbols referenced in the construction of its types (think `@NgModule`, `@Component`, `@Injectable`, etc.) to be statically defined. For this reason, we cannot dynamically inject state at runtime with AoT unless we provide `initialState` as a function. Thus the above `NgModule` definition simply changes to:
+
+```ts
+/// Pretend this is dynamically injected at runtime
+const initialStateFromSomewhere = { counter: 3 };
+
+/// Static state
+const initialState = { counter: 2 };
+
+/// In this function dynamic state slices, if they exist, will overwrite static state at runtime.
+const getInitialState = () => ({...initialState, ...initialStateFromSomewhere});
```

Comment:
This should be exported and a plain function instead of a arrow function for AOT support